internal/proxyservice/tcp/mysql/plugin.go

```diff
@@ -0,0 +1,44 @@
+package mysql
+
+import (
+	"net"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+// NewConnector is a required method on the tcp.Plugin interface. It returns a
+// tcp.Connector.
+//
+// The single argument passed in is of type connector.Resources. It contains
+// connector-specific config and a logger.
+func NewConnector(conRes connector.Resources) tcp.Connector {
+	return func(
+		clientConn net.Conn,
+		credentialValuesByID connector.CredentialValuesByID,
+	) (backendConn net.Conn, err error) {
+		// create a connector on a per connection basis
+		connConnector := &Connector{
+			logger:   conRes.Logger(),
+		}
```

Comment:
How about `singleUseConn := SingleUseConnector...`? (per discussion above?)

Conceptually there's a lot going on here and I'd like to help the reader out as much as possible.